src/argilla/training/base.py

```diff
@@ -205,3 +228,39 @@ def save(self, output_dir: str):
           output_dir (str): The path to the directory where the model will be saved.
         """
         self._trainer.save(output_dir)
+
+
+class ArgillaTrainerSkeleton(object):
```

Comment:
```suggestion
class ArgillaTrainerSkeleton(object):
```
If you want, you can turn this into an ABC subclass. Then you can use `@abstractmethod` before the yet-to-be-implemented methods, and you don't need the raise NotImplementedError, as any class that has an abstract method that wasn't overridden can't be initialized.
The abc stuff is just from the Python standard library